DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs

```diff
@@ -535,28 +535,46 @@ protected void ListMaps()
             lbGameModeMapList.SelectedIndex = -1;
 
             int mapIndex = -1;
-            int skippedMapsCount = 0;
 
             var isFavoriteMapsSelected = IsFavoriteMapsSelected();
             var maps = GetSortedGameModeMaps();
 
             bool gameModeMapChanged = false;
 
-            for (int i = 0; i < maps.Count; i++)
+            List<GameModeMap> filteredMaps;
+
+            if (tbMapSearch.Text != tbMapSearch.Suggestion)
             {
-                var gameModeMap = maps[i];
-                if (tbMapSearch.Text != tbMapSearch.Suggestion)
+                var search = tbMapSearch.Text.ToUpperInvariant();
+                var searchWords = search.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+                // exact match
+                var exactMatches = maps.Where(gmm =>
+                    gmm.Map.Name.ToUpperInvariant().Contains(search) ||
+                    gmm.Map.UntranslatedName.ToUpperInvariant().Contains(search)).ToList();
+
+                // matches with "AND" logic: Word1 AND Word2 AND Word3
+                var partialMatches = maps.Except(exactMatches).Where(gmm =>
                 {
-                    string promptUpper = tbMapSearch.Text.ToUpperInvariant();
-                    bool mapMatches = gameModeMap.Map.Name.ToUpperInvariant().Contains(promptUpper)
-                        || gameModeMap.Map.UntranslatedName.ToUpperInvariant().Contains(promptUpper);
+                    bool allInTranslated = searchWords.All(word =>
+                        gmm.Map.Name.ToUpperInvariant().Contains(word));
 
-                    if (!mapMatches)
-                    {
-                        skippedMapsCount++;
-                        continue;
-                    }
-                }
+                    bool allInUntranslated = searchWords.All(word =>
+                        gmm.Map.UntranslatedName.ToUpperInvariant().Contains(word));
+
+                    return allInTranslated || allInUntranslated;
+                }).ToList();
+
+                filteredMaps = exactMatches.Concat(partialMatches).ToList();
```

Comment:
you can use `filteredMaps = [..exactMatches, ..partialMatches]` for a better readability. An optional advise